### ISSUE TITLE
Item Stacking Issue

### ISSUE THREAD

**Issue by AMinecraftDev** [tastybento/ASkyBlock-Bugs-N-Features#616]
Thank you for filing a bug report. Please complete these sections to help speed resolution.

**Server version:**
1.12.1

**Plugin version:**
V3.0.8.2

**What other plugins are you using? (do /plugins and paste here)**
https://gyazo.com/eb53c4decc79375830b1aa89692428b7

**Description of issue. What happened?**
People are able to stack the item from their inventory and stack it into the one in their hand, even if the stack is over the max stack limit.

**Steps to make this happen**
https://youtu.be/lKC3fa9hpsU

**What do you think should happen?**
Fix it please :D

**Comment by Poslovitch** [comment]
Thanks for the report :)

Can you please provide the following information : 
- what are the askyblock permissions of the player who did the video?
- is he op?
- what are the settings of the concerned island? (i'm looking for VISITOR_ITEM_DROP and VISITOR_ITEM_PICKUP)

**Comment by AMinecraftDev** [comment]
**What are the askyblock permissions of the player who did the video?**
I just tested it also, I set myself to Mediocre rank on my server (non-staff, non-op) and I could replicate this. The aSkyblock permissions I have when I'm this rank are attached to this link: https://hastebin.com/enokamapeq.css

**Is he op?** 
Yes he is, but it works also when you're non-op, as I just tested it.

**What are the settings of the concerned island?**
VISITOR_ITEM_DROP and VISITOR_ITEM_PICKUP are both set to false.

**Comment by tastybento** [comment]
Hi - sorry for the late response.

You wrote:
People are able to stack the item from their inventory and stack it into the one in their hand, even if the stack is over the max stack limit.

I don't understand this problem. What is the "max stack limit"? There isn't such a limit in ASkyBlock.

**Comment by Poslovitch** [comment]
@tastybento You should  have a look at the video.
Basically, there is this method in Bukkit : `ItemStack#setMaxStackSize(int i);`
This method helps to reduce the maximum stack size (which is by default 64), or to make it higher (e.g with eggs, whose default is 16).

The thing I don't understand though, is that we never use such method in ASB...

**Comment by tastybento** [comment]
@poslovich yeah, I watched the video and I still don’t understand how this is related to Askyblock. I don’t understand the problem even. What is happening that should not be happening? All I see is a load of swords.

**Comment by AMinecraftDev** [comment]
I see what you mean about how it couldn't be aSkyblock issue, but if you load in aSkyblock on a server alone and try it the issue is still present. I possibly think it's how you handle your event cancelling for when someone is on someone else's island.

**Comment by tastybento** [comment]
I can't replicate this on 1.12.2. You say you are running server version 1.12.1, but your client is not 1.12 - there's no off-hand or recipe book in the UI, so maybe this is a via version thing.
